arm/proofs/arm.ml

```diff
@@ -740,10 +756,13 @@ let ARM_MACRO_SIM_ABBREV_TAC =
     fun n ->  if n = 0 then pat0
               else vsubst[mk_small_numeral n,pan] patn
   and grab_dest =
-    let pat = `read (memory :> bytes(p,8 * n))` in
+    let pat = `read (memory :> bytes(p,8 * n)) s` in
```

Comment:
This part was updated because sometimes `ARM_MACRO_SIM_ABBREV_TAC` was grabbing the wrong `read (memory :> bytes ...)` expression from the postcondition and deciding that was the output buffer, such as `read (memory :> bytes ...) s_input` (where `s_input` is some imaginary input state). I resolved this solution by explicitly stating that it must use a variable `s:armstate` where `s` is the variable provided by the lambda function of the post condition. Also updated at line 742 that `s` must be the postcondition's state variable.